samples/system-test/detect.v2beta1.test.js

```diff
@@ -0,0 +1,148 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const test = require(`ava`);
+const {runAsync} = require('@google-cloud/nodejs-repo-tools');
+
+const cmd = 'node detect.v2beta1.js';
+//const testQuery = [`How do I sign up?`];
+const testQuery = `Where is my data stored?`;
+
+const testKnowledgeBaseName = 'TestKnowBase';
+
+let knowbaseDisplayName;
+let knowbaseFullName;
+
+let documentFullPath;
+const testDocName = `TestDoc`;
+//const testDocumentPath = `https://dialogflow.com/docs/concepts/google-projects-faq`;
+const testDocumentPath = `https://cloud.google.com/storage/docs/faq`;
+
+test.before(`It should create a knowledge base`, async t => {
+  const output = await runAsync(
+    `${cmd} createKnowledgeBase -k ${testKnowledgeBaseName}`
+  );
+  const parsedOut = output.split(`\n`);
+  const confirmName = parsedOut[1].split(`:`)[1].trim();
+  t.true(confirmName === testKnowledgeBaseName);
+  knowbaseDisplayName = confirmName;
+  knowbaseFullName = parsedOut[0].split(`:`)[1].trim();
+});
+
+test.before(`It should detect Intent with Model Selection`, async t => {
+  const output = await runAsync(`${cmd} detectIntentwithModelSelection`);
+  t.true(
+    output.includes(
+      `I can help with that. Where would you like to reserve a room?`
+    )
+  );
+});
+
+test.serial(
+  `It should register newly created knowledge base with getKnowledgeBase`,
+  async t => {
+    const output = await runAsync(
+      `${cmd} getKnowledgeBase -n "${knowbaseFullName}"`
+    );
+    //const parsedOut = output.split(`\n`);
+    t.true(
+      output.includes(knowbaseFullName) && output.includes(knowbaseDisplayName)
+    );
+  }
+);
+
+test.serial(
+  `It should locate knowledge base with listKnowledgeBases`,
+  async t => {
+    const output = await runAsync(`${cmd} listKnowledgeBases`);
+    t.true(output.includes(testKnowledgeBaseName));
+    const parsedOut = output.split(`\n`);
+    parsedOut.forEach(p => {
+      if (p.includes(testKnowledgeBaseName)) t.pass();
+    });
+  }
+);
+
+test.serial(`It should create a document in Knowledge Base`, async t => {
+  const output = await runAsync(
+    `${cmd} createDocument -n "${knowbaseFullName}" -z "${testDocumentPath}" -m "${testDocName}"`
+  );
+  t.true(output.includes(`Document created`));
+});
+
+test.serial(
+  `It should list document just created with listDocuments`,
+  async t => {
+    const output = await runAsync(
+      `${cmd} listDocuments -n "${knowbaseFullName}"`
+    );
+    const parsedOut = output.split(`\n`);
+    documentFullPath = parsedOut[parsedOut.length - 1].split(`:`)[1];
+    t.true(documentFullPath !== ``);
```

Comment:
❗️ **#BLOCKER**

Checking that something is not an empty string is not a valid system test assertion